packages/angular_devkit/build_angular/src/browser/index.ts

```diff
@@ -592,34 +595,55 @@ export function buildWebpackBrowser(
               }
 
               let bundleInfoText = '';
-              const processedNames = new Set<string>();
-              for (const result of processResults) {
-                processedNames.add(result.name);
-
-                const chunk =
-                  webpackStats &&
-                  webpackStats.chunks &&
-                  webpackStats.chunks.find(c => result.name === c.id.toString());
-                if (result.original) {
-                  bundleInfoText +=
-                    '\n' + generateBundleInfoStats(result.name, result.original, chunk);
-                }
-                if (result.downlevel) {
-                  bundleInfoText +=
-                    '\n' + generateBundleInfoStats(result.name, result.downlevel, chunk);
-                }
-              }
-
               if (webpackStats && webpackStats.chunks) {
                 for (const chunk of webpackStats.chunks) {
-                  if (processedNames.has(chunk.id.toString())) {
-                    continue;
+                  const result = processResults.find((result) => result.name === chunk.id.toString());
+                  if (result) {
+                    if (result.original) {
+                      bundleInfoText +=
+                        '\n' + generateBundleInfoStats(result.name, result.original, chunk);
+                    }
+                    if (result.downlevel) {
+                      bundleInfoText +=
+                        '\n' + generateBundleInfoStats(result.name, result.downlevel, chunk);
+                    }
+                  } else {
+                    const asset =
+                    webpackStats.assets && webpackStats.assets.find(a => a.name === chunk.files[0]);
+                    bundleInfoText +=
+                      '\n' + generateBundleStats({ ...chunk, size: asset && asset.size }, true);
                   }
+                }
+              }
 
-                  const asset =
-                    webpackStats.assets && webpackStats.assets.find(a => a.name === chunk.files[0]);
-                  bundleInfoText +=
-                    '\n' + generateBundleStats({ ...chunk, size: asset && asset.size }, true);
+              const budgets = options.budgets || [];
+              for (const budget of budgets) {
+                const sizes = calculateSizes(budget, webpackStats);
+                for (const threshold of calculateThresholds(budget)) {
+                  for (const {size, label} of sizes) {
+                    if (exceededThreshold(size, threshold)) {
+                      const colorize = colorizeSeverity(threshold.severity);
+                      const severity = stringifySeverity(threshold.severity);
+
+                      switch (threshold.type) {
+                        case ThresholdType.MAX: {
```

Comment:
Moving the threshold checks into these cases would reduce the block depth by 1 and remove the need for the `exceededThreshold` function.
Maybe something like for max:
```
if (size <= threshold.limit) {
  continue;
}
```

Reply:
Good point. Done.